boa/profiling.py

```diff
@@ -150,11 +150,12 @@ def by_line(self):
         for pc in self.computation.code._trace:
             if (node := source_map.get(pc)) is not None:
                 current_line = node.lineno
+                filepath = node.module_node.path
 
-            # NOTE: do we still need the `current_line is not None` guard?
-            if current_line is not None and pc not in seen:
-                ret.setdefault(current_line, Datum())
-                ret[current_line].merge(self.by_pc[pc])
+            # TODO hm, maybe we should not actually use this guard?
```

Comment:
why was it there on the 1st place? :thinking:

Reply:
explained here: https://github.com/vyperlang/titanoboa/pull/282/files#diff-5a474938e5fc1106ddaceed0d36987ec8b2f03b925965cc78df01bfff3c8d3c2R150-R153